src/utils.c

```diff
@@ -275,35 +275,73 @@ get_interval_period_timestamp_units(Interval *interval)
 #endif
 }
 
+#ifdef HAVE_INT64_TIMESTAMP
+#define JAN_3_2000 (2 * USECS_PER_DAY)
+#else
+#define JAN_3_2000 (2 * SECS_PER_DAY)
+#endif
+
+/*
+* The default origin is Monday 2000-01-03. We don't use PG epoch since it starts on a saturday.
+* This makes time-buckets by a week more intuitive and aligns it with
+* date_trunc.
+*/
+#define DEFAULT_ORIGIN (JAN_3_2000)
+#define TIME_BUCKET_TS(period, timestamp, result, shift)			\
+	do															\
+	{															\
+		if (TIMESTAMP_NOT_FINITE(timestamp)) 					\
+			PG_RETURN_TIMESTAMP(timestamp);						\
```

Comment:
Does this strictly need to be a macro given that we are always passing in a timestamp? Or do we need to treat `Timestamp` and `TimestampTz` differently?